### PR TITLE
Port grouping redo/v17.2

### PR DESCRIPTION
**Pull request by victorjulien**
#10497 with score based sorting

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_autofp_suri_time.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 101 | 112 | 110.89% | 

Pipeline 18726

File: src/detect-engine-build.c [hunk 1]
```diff
@@ -1139,6 +1139,14 @@ typedef enum {
...
 static int CreateGroupedPortListCmpCnt(const DetectPort *a, const DetectPort *b)
 {
+    if (a->sh->init->score < b->sh->init->score) {
+        return 1;
+    } else if (a->sh->init->score > b->sh->init->score) {
+        return -1;
+    }
+
+    /* if scores are equal, check sig cnt */
+
```
Comment (inashivb) [inline src/detect-engine-build.c:1149]: This is weird. The list at this point should only belong to one particular score as a part of the optimizations I did earlier. 🤔  Checking..

Comment (inashivb) [reply src/detect-engine-build.c:1149]: There was a typo in what I had done. 🤦🏽‍♀️ 
Fixing that is enough and we don't need to check the scores again. :) See https://github.com/OISF/suricata/pull/10500

**Comment by victorjulien** [comment]
replaced by #10500

**Pull request closed without merging**